src/main/java/pl/antonina/tasks/allegro/accessToken/AccessTokenProvider.java

```diff
@@ -0,0 +1,39 @@
+package pl.antonina.tasks.allegro.accessToken;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.*;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+@Service
+public class AccessTokenProvider {
+
+    private final OkHttpClient httpClient = new OkHttpClient.Builder().build();
+
+    public String getAccessToken() {
+        RequestBody formBody = new FormBody.Builder().build();
+        String clientId = "2d28293921ae472c9d82d99b2d9268eb";
```

Comment:
As a separate task: move these IDs, URLs and so on to a config/properties file. :) It can be useful even when testing! https://www.baeldung.com/spring-profiles